pilot/pkg/config/kube/ingress/controller_test.go

```diff
@@ -201,26 +195,20 @@ func TestIngressController(t *testing.T) {
 	}
 
 	// Create a "real" ingress resource, with 4 host/path rules and an additional "default" rule.
-	const expectedRuleCount = 5
 	if _, err := cl.ExtensionsV1beta1().Ingresses(namespace).Create(&ingress); err != nil {
 		t.Errorf("Cannot create ingress in namespace %s (error: %v)", namespace, err)
 	}
 
-	util.Eventually(func() bool {
-
-		lock.Lock()
-		defer lock.Unlock()
-
-		return notificationCount == expectedRuleCount
-	}, t)
-	if notificationCount != expectedRuleCount {
-		t.Errorf("expected %d IngressRule events to be notified, found %d", expectedRuleCount, notificationCount)
+	const expectedRuleCount = 5
+	actual := 0
+	for {
```

Comment:
If we get less than 5 events this loop waits for ever. What was the problem with eventually?

Reply:
This is like adding “wait indefinitely parameter” to eventually

Reply:
let the loop to run when there is a new event instead of time interval is a better model. I can add a timeout if it's really necessary for this test otherwise we can rely on the timeout from test framework.